cms/djangoapps/contentstore/tasks.py

```diff
@@ -270,8 +270,15 @@ def create_export_tarball(course_module, course_key, context, status=None):
 
     Updates the context with any error information if applicable.
     """
+    export_path = settings.MEDIA_ROOT / 'exports'
```

Comment:
This is potentially problematic and worth discussion. Currently we are creating these temporary export files in /tmp then using Django Storages to move them to the configured semi-permanent download location (S3 for edx.org). Django 1.11 has enhanced security to prevent it from allowing FileFields to access files from anywhere outside of settings.MEDIA_ROOT, which is the publicly accessible upload path. As such this change makes these export files available in that directory path during creation (albeit in randomly named files). Mitigation options for this off-hand include:

- Locking down security on the MEDIA_ROOT / exports url, which is likely to be error prone and forgotten by partners.
- Writing a custom FileField implementation that lets us override the security feature to use a different path.

Reply:
Where exactly is it hitting the problem?  From my understanding, the current workflow should have been something like this:

1. Export tarball is created using generic Python code under /tmp
1. Django is told the filename and given the content of the file
1. Django copies it to an appropriate location for the current storage system (which is under MEDIA_ROOT for the default one) when saving the model

Reply:
It happens in [step 2](https://github.com/edx/edx-platform/blob/113abb5bff1bb0299adbc505c67c13802f0a9495/cms/djangoapps/contentstore/tasks.py#L257), when the contents of the file are read into a Django File for the UserTaskArtifact.

Reply:
Looking at one of the failures for that, I wonder if we're sending it "/tmp/filename.tar.gz" as the filename, which is tripping up the join operation and just ignoring the intended path prefix.  Could you try extracting just the filename from "tarball.name" on that line, and pass that in as the first argument to `save()`?

Reply:
Seems to have worked, thanks!